flink-runtime/src/main/java/org/apache/flink/runtime/rest/FileUploadHandler.java

```diff
@@ -72,10 +82,12 @@ public FileUploadHandler(final Path uploadDir) {
 	protected void channelRead0(final ChannelHandlerContext ctx, final HttpObject msg) throws Exception {
 		if (msg instanceof HttpRequest) {
 			final HttpRequest httpRequest = (HttpRequest) msg;
+			LOG.trace("Received request. URL:{} Method:{}", httpRequest.getUri(), httpRequest.getMethod());
 			if (httpRequest.getMethod().equals(HttpMethod.POST)) {
 				if (HttpPostRequestDecoder.isMultipart(httpRequest)) {
 					currentHttpPostRequestDecoder = new HttpPostRequestDecoder(DATA_FACTORY, httpRequest);
 					currentHttpRequest = httpRequest;
+					currentUploadDir = Files.createDirectory(uploadDir.resolve(UUID.randomUUID().toString()));
```

Comment:
How do we clean up the `currentUploadDir` in case of a failure?

Reply:
If the `FileUploadHandler` itself fail it isn't cleaned up, but that was already the case in the existing code. The handler is generally rather _light_ when it comes to failure handling (i.e. it doesn't do anything in that regard).

Reply:
I think we should add this.